nixos/modules/services/networking/ssh/sshd.nix

```diff
@@ -128,6 +128,16 @@ in
         '';
       };
 
+      firewallOpenPorts = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Specifies whether the ports the SSH daemon listens on must
+          be enabled in the firewall. Remember to open at least some
+          of them.
```

Comment:
@P-E-Meunier Yeah, that's why changing the default is indeed really problematic...

However using `services.openssh.listenAddresses` with a different port than  `services.openssh.ports` will still lock you out (but obviously you already know that).

Another possible solution would be to use your new option for allowing the (IP, Port) combinations from `services.openssh.listenAddresses` as well, which would still be against the convention but at least improve the current situation.

@edolstra, @Mic92, @wkennington, #19504 - Any opinions on that?

We could also start using the "services" solution (https://github.com/NixOS/nixpkgs/issues/19504#issuecomment-271147629 from @Mic92) which would also greatly improve our current situation (imho) but probably require some additional thinking/discussion.

Imho it would also be a good idea to move that option over to the firewall module (e.g. `networking.firewall.allowSSH` (or `allowServices = [ ... ]`, `allowPorts = [ service.openssh.ports ]`) - Could probably even be the best solution (idk)...